src/pages/user/authority/AuthorityOrcidModal.tsx

```diff
@@ -1,22 +1,24 @@
-import React, { FC, useState } from 'react';
-import { useSelector } from 'react-redux';
+import React, { useState } from 'react';
 
 import AuthorityModal from './AuthorityModal';
 import OrcidModal from './OrcidModal';
-import { RootStore } from '../../../redux/reducers/rootReducer';
+import { User } from '../../../types/user.types';
 
 enum ModalType {
   AUTHORITY = 'authority',
   NONE = '',
   ORCID = 'orcid',
 }
 
-const AuthorityOrcidModal: FC = () => {
-  const { authority } = useSelector((store: RootStore) => store.user);
-  const [openModal, setOpenModal] = useState(authority ? ModalType.NONE : ModalType.AUTHORITY);
+interface AuthorityOrcidModalProps {
+  user: User;
+}
+
+const AuthorityOrcidModal = ({ user }: AuthorityOrcidModalProps) => {
+  const [openModal, setOpenModal] = useState(user.authority ? ModalType.NONE : ModalType.AUTHORITY);
 
   const handleNextClick = () => {
-    if (authority && authority.orcids.length === 0) {
+    if (user.authority && user.authority.orcids.length === 0) {
```

Comment:
dette blir det samme(?)
```suggestion
    if (user.authority?.orcids.length === 0) {
```